### PR TITLE
[Issue #3] Feature Sending Images

### PR DESCRIPTION
**Pull request by jpedrosiq**
# Motivation

- We need to add the functionality of sending messages in the chat, by uploading files that will be hosted via Cloudinary.
- Link to issue: https://github.com/jpedrosiq/messenger-app/issues/8

# Changes

- Add button in Chat field that uploads images
- Add overflow-Y to both Conversations and Chat
- Add Logout button as "..."
- Alert that says when pictures were successfully uploaded
- Scroll to bottom functionality when conversation is open and when sending messages

# Screenshots of feature to send pictures

### Single picture with text

image

### Multiple images + Single images with no text

image

### CODE REVIEW

**Review by cassandrale179 — COMMENTED** [review]
Looks good, only left 1 comment to address!

File: client/src/components/ActiveChat/Header.js [hunk 1]
```diff
@@ -10,39 +10,34 @@ const useStyles = makeStyles((theme) => ({
...
     height: 89,
     marginBottom: 34,
-    boxShadow: "0 2px 20px 0 rgba(88,133,196,0.10)"
+    boxShadow: "0 2px 20px 0 rgba(88,133,196,0.10)",
   },
   content: {
     display: "flex",
     alignItems: "center",
-    marginLeft: 24
+    marginLeft: 24,
```
Comment (FilipeGorodscy) [inline client/src/components/ActiveChat/Header.js:18]: You could use `theme.spacing()` to calculate margin/padding with Material UI

File: client/src/components/ActiveChat/Header.js [hunk 2]
```diff
@@ -10,39 +10,34 @@ const useStyles = makeStyles((theme) => ({
...
-    letterSpacing: -0.17
+    letterSpacing: -0.17,
   },
   statusDot: {
     height: 8,
     width: 8,
     borderRadius: "50%",
     marginRight: 5,
-    backgroundColor: "#D0DAE9"
+    backgroundColor: "#D0DAE9",
```
Comment (FilipeGorodscy) [inline client/src/components/ActiveChat/Header.js:36]: You can use `theme.palette` for colours in Material UI

File: client/src/components/ActiveChat/Header.js [hunk 3]
```diff
@@ -53,10 +48,16 @@ const Header = (props) => {
     <Box className={classes.root}>
       <Box className={classes.content}>
         <Typography className={classes.username}>{username}</Typography>
-        <Box className={`${classes.statusDot} ${classes[online && "online"]}`}></Box>
-        <Typography className={classes.statusText}>{online ? "Online" : "Offline"}</Typography>
+        <Box
+          className={`${classes.statusDot} ${classes[online && "online"]}`}
+        ></Box>
```
Comment (FilipeGorodscy) [inline client/src/components/ActiveChat/Header.js:53]:
```suggestion
        <Box className={`${classes.statusDot} ${classes[online && "online"]}`} />
```

File: client/src/components/ActiveChat/Input.js [hunk 4]
```diff
@@ -39,11 +53,56 @@ class Input extends Component {
...
+    const formData = new FormData();
+    const attachments = [];
+
+    // function that handles each promise post request from Cloudinary
+    const promiseForAttachmentUrl = (attachment) => {
+      formData.append("file", attachment);
+      formData.append("upload_preset", "vgnox97i");
+
+      return fetch("https://api.cloudinary.com/v1_1/dtcgl7plw/image/upload", {
+        method: "POST",
```
Comment (FilipeGorodscy) [inline client/src/components/ActiveChat/Input.js:82]: I'd suggest hiding the `upload_preset` and `cloud_name` inside `.env`.

File: client/src/components/ActiveChat/Input.js [hunk 5]
```diff
@@ -39,11 +53,56 @@ class Input extends Component {
...
+      .then((dataArr) => {
+        dataArr.forEach((data) => {
+          // only push the attachment to attachments array if url exists
+          data.url &&
+            this.setState((prevState) => ({
+              attachments: [data.url, ...prevState.attachments],
+            }));
+        });
+        alert("Attachments successfully uploaded! Press 'enter' to submit.");
+      });
```
Comment (FilipeGorodscy) [inline client/src/components/ActiveChat/Input.js:105]: You should use the `async` `await` keywords to be consistent with the rest of the codebase. See [this document](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Async_await#awaiting_a_promise.all) for reference :)

File: client/src/components/ActiveChat/SenderBubble.js [hunk 6]
```diff
@@ -1,38 +1,67 @@
...
-    marginBottom: 5
+    marginBottom: 5,
   },
   text: {
     fontSize: 14,
     color: "#91A3C0",
     letterSpacing: -0.2,
     padding: 8,
-    fontWeight: "bold"
+    fontWeight: "bold",
```
Comment (FilipeGorodscy) [inline client/src/components/ActiveChat/SenderBubble.js:24]: You should use `theme.typography` for fonts in Material UI :)

File: client/src/components/LogoutButton.js [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    const { classes } = this.props;
+    if (!this.props.user.id) {
+      // If we were previously logged in, redirect to login instead of register
+      if (this.state.isLoggedIn) return <Redirect to="/login" />;
+      return <Redirect to="/register" />;
+    }
+    return (
+      <>
+        {/* logout button will eventually be in a dropdown next to username */}
+        {/* <MoreHorizIcon classes={{ root: classes.ellipsis }} /> */}
```
Comment (FilipeGorodscy) [inline client/src/components/LogoutButton.js:47]: I think this is from another PR, but make sure to remove any commented code

**Review by FilipeGorodscy — APPROVED** [review]
Looks good to me :) I left some small comments. Feel free to merge after you make the changes 👍

**Pull request merged**